ui/components/src/Table/Table.stories.tsx

```diff
@@ -0,0 +1,434 @@
+    data: generateMockTableData(500),
+    columns: COLUMNS,
+    checkboxSelection: true,
+    getCheckboxColor: (data) => data.color,
+  },
+  argTypes: {
+    // Disabling values managed within the render below.
+    onRowSelectionChange: {
+      table: {
+        disable: true,
+      },
+    },
+    rowSelection: {
+      table: {
+        disable: true,
+      },
+    },
+    checkboxSelection: {
+      table: {
+        disable: true,
+      },
+    },
+  },
+  render: (args) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [rowSelection, setRowSelection] = useState<TableProps<MockTableData>['rowSelection']>({});
+
+    const handleRowSelectionChange: TableProps<MockTableData>['onRowSelectionChange'] = (newRowSelection) => {
+      action('onRowSelectionChange')(newRowSelection);
+      setRowSelection(newRowSelection);
+    };
+
+    return <Table {...args} onRowSelectionChange={handleRowSelectionChange} rowSelection={rowSelection} />;
+  },
+};
+
+/**
+ * Currently, a table with no data shows the header with no rows. This UI may
+ * change in the future.
+ */
+export const NoData: Story = {
+  args: {
+    data: [],
+    columns: COLUMNS,
+  },
+};
+
+/**
+ * The table supports some basic common keyboard navigation to support
+ * keyboard-based interactions with the table including:
+ * - Initial focus on top left cell.
+ * - right arrow: moves the focus one cell to the right
+ * - left arrow: moves the focus one cell to the left
+ * - up arrow: moves the focus one cell up
+ * - down arrow: moves the focus one cell down
+ * - home: moves focus to the top left cell
+ * - end: moves focus to the bottom right cell
+ * - page up: moves the focus up approximately one page
+ * - page down: moves the focus down approximately one page
+ */
+export const KeyboardNavigation: Story = {
+  args: {
+    data: generateMockTableData(500),
+    columns: COLUMNS,
+    checkboxSelection: true,
+    getCheckboxColor: (data) => data.color,
+  },
+  argTypes: {
+    // Disabling values managed within the render below.
+    onRowSelectionChange: {
+      table: {
+        disable: true,
+      },
+    },
+    rowSelection: {
+      table: {
+        disable: true,
+      },
+    },
+    checkboxSelection: {
+      table: {
+        disable: true,
+      },
+    },
+  },
+  parameters: {
+    // This story exists to test for interactions. No screenshots.
+    happo: false,
+  },
+  render: (args) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [rowSelection, setRowSelection] = useState<TableProps<MockTableData>['rowSelection']>({});
+
+    const handleRowSelectionChange: TableProps<MockTableData>['onRowSelectionChange'] = (newRowSelection) => {
+      action('onRowSelectionChange')(newRowSelection);
+      setRowSelection(newRowSelection);
+    };
+
+    return <Table {...args} onRowSelectionChange={handleRowSelectionChange} rowSelection={rowSelection} />;
+  },
+  play: async ({ canvasElement }) => {
```

Comment:
This provides us with an [interaction test](https://storybook.js.org/docs/react/writing-tests/interaction-testing) for the keyboard navigation.